### PR TITLE
Test Puppeteer in Jenkins

### PR DESCRIPTION
**Pull request by Nasir2712**

### CODE REVIEW

**Comment by Nasir2712** [comment]
jenkins, retest this please

**Comment by Nasir2712** [comment]
retest this please

**Comment by SashaKyz** [comment]
retest this please

**Comment by SashaKyz** [comment]
retest this please

File: tools/srcServer.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ const bundler = webpack(config)
 // Run Browsersync and use middleware for Hot Module Replacement
 browserSync({
   port: 3000,
+  open: false,
```
Comment (ratneshraval) [inline tools/srcServer.js]: Will not open browser automatically on `yarn start` better setup for container.

File: package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     "build:preprod": "ENV_PATH=./env/.preprod.env babel-node tools/build.js && npm run open:dist",
     "build:production": "ENV_PATH=./env/.production.env babel-node tools/build.js && npm run open:dist",
     "license-dump": "license-checker --csv --out legal/licenses.csv",
-    "test": "jest",
+    "test": "jest --detectOpenHandles",
```
Comment (ratneshraval) [inline package.json]: Helpful to detect when tests are done but command doesn't finish.

File: jest-puppeteer.config.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  launch: {
+    dumpio: true,
+    headless: true,
+    devtools: false,
+    args: ['--no-sandbox', '--disable-setuid-sandbox']
+  },
+  server: {
+    command: 'ENV_PATH=./env/.integration.env npm run open:src',
+    launchTimeout: 10000,
+    port: 3000
```
Comment (ratneshraval) [inline jest-puppeteer.config.js]: Important to match port numbers. We need to put it somewhere as single variable.

**Comment by sunilb0575** [comment]
jenkins, retest this please

**Pull request merged**